base_layer/core/src/transactions/transaction.rs

```diff
@@ -353,7 +394,38 @@ impl TransactionOutput {
 
     /// Verify that range proof is valid
     pub fn verify_range_proof(&self, prover: &RangeProofService) -> Result<bool, TransactionError> {
-        Ok(prover.verify(&self.proof.to_vec(), &self.commitment))
+        Ok(prover.verify(&self.proof.0, &self.commitment))
+    }
+
+    /// Attempt to rewind the range proof to reveal the proof message and committed value
+    pub fn rewind_range_proof(
```

Comment:
```suggestion
    pub fn rewind_range_proof_value_only(
```
Not sure what a better name would be, but my intention here is to verbalise how this method differs from `full_rewind_range_proof`